redis.go

```diff
@@ -299,7 +299,14 @@ func (c *baseClient) initConn(ctx context.Context, cn *pool.Conn) error {
 		// difficult to rely on error strings to determine all results.
 		return err
 	}
-
+	if !c.opt.DisableIndentity {
```

Comment:
Please move the code after line 318. When protocol is 2, the application will get NOAUTH error since AUTH has not requested.